drivers/net/ethernet/intel/idpf/eth/idpf_ethtool.c

```diff
@@ -131,23 +146,27 @@ static int idpf_set_rxfh(struct net_device *netdev,
 			 struct netlink_ext_ack *extack)
 {
 	struct idpf_netdev_priv *np = netdev_priv(netdev);
+	struct idpf_eth_adapter *adapter;
 	struct idpf_rss_data *rss_data;
-	struct idpf_adapter *adapter;
 	struct idpf_vport *vport;
 	int err = 0;
 	u16 lut;
 
 	idpf_vport_ctrl_lock(netdev);
-	vport = idpf_netdev_to_vport(netdev);
+	if (test_bit(IDPF_ETH_RESET_IN_PROG, np->adapter->flags)) {
+		err = -EBUSY; /* Temporarily busy indication */
+		goto unlock_mutex;
+	}
 
+	vport = idpf_netdev_to_vport(netdev);
 	adapter = vport->adapter;
-
-	if (!idpf_is_cap_ena_all(adapter, IDPF_RSS_CAPS, IDPF_CAP_RSS)) {
+	if (!idpf_is_cap_ena_all(&adapter->dev_info->caps,
```

Comment:
use a macro like idpf_caps(adapter) to go from adapter to caps pointer. This is necessary for consistency and to guard against future churn due to review comments.

Reply:
Whenever you are using 2 or more levels of pointer indirection, it is cleaner and robust to use a macro. It guards against code review comments also because you don't want to end up modifying 15 usages of caps vs one change in a header file.

Reply:
Okay, I will add the macro.

Reply:
Done!